Mathlib/LinearAlgebra/InvariantBasisNumber.lean

```diff
@@ -21,14 +22,29 @@ that a surjective linear map `(Fin n → R) →ₗ[R] (Fin m → R)` implies `m
 the strong rank condition, that an injective linear map `(Fin n → R) →ₗ[R] (Fin m → R)`
 implies `n ≤ m`.
 
+Orzech, Djoković and Ribenboim considered the following property for a ring `R`:
+for any finitely generated `R`-module `M`, any surjective homomorphism `f : N → M`
+from a submodule `N` of `M` to `M` is injective.
+It's called `Π`-ring or `Π₁`-ring in their papers.
+They showed that any Noetherian ring (not necessarily commutative) and any commutative ring
+satisfies this property.
+It's easy to see that such property implies strong rank condition given the ring is non-trivial.
+
 The strong rank condition implies the rank condition, and the rank condition implies
 the invariant basis number property.
+The converse is not true, as we have counterexamples in the book of Lam
```

Comment:
Add bibtex reference to Lam

Reply:
I'm confused with the descriptions in <https://leanprover-community.github.io/contribute/doc.html#citing-other-works>. It gives three examples in the end, but seems that none of references in them works. The reference page for mathlib docs is just 404.

Reply:
I don't know why the references pages are not visible in mathlib4_docs.
To start with, I believe it is enough to add the references of Ribenboim, Orzech and Lam to docs/reference.bib and to mention them with the adequate label.

Reply:
See : https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/references.2Ebib/near/338473634

Reply:
Added.